packages/components/src/Modal/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,183 @@
+import React from "react"
+import classnames from "classnames"
+import {
+  Assertive,
+  Cautionary,
+  Informative,
+  Negative,
+  Positive,
+} from "@kaizen/draft-illustration"
+import { ButtonProps } from "~components/Button"
+import { Heading } from "~components/Heading"
+import {
+  CautionWhiteIcon,
+  ExclamationWhiteIcon,
+  InformationWhiteIcon,
+  SuccessWhiteIcon,
+} from "~components/Icon"
+import {
+  GenericModal,
+  ModalDescription,
+  ModalFooter,
+  ModalHeader,
+  ModalLabel,
+  ModalSection,
+} from "~components/Modal/GenericModal"
+import styles from "./ConfirmationModal.module.scss"
+
+export type ConfirmationModalProps = {
+  isOpen: boolean
+  unpadded?: boolean
+  /**
+   * To display the Prominent variation of the modal types
+   */
+  isProminent?: boolean
+  /**
+   * Defines the modal type or mood
+   */
+  mood: Mood
+  title: string
+  onConfirm?: () => void
+  onDismiss: () => void
+  onAfterLeave?: () => void
+  confirmLabel?: string
+  dismissLabel?: string
+  confirmWorking?: { label: string; labelHidden?: boolean }
+  automationId?: string
+  children: React.ReactNode
+}
+
+type Mood = "positive" | "informative" | "negative" | "cautionary" | "assertive"
+
+const getIcon = (mood: Mood, isProminent: boolean): JSX.Element => {
+  switch (mood) {
+    case "cautionary":
+      return isProminent ? (
+        <Cautionary />
+      ) : (
+        <CautionWhiteIcon inheritSize role="presentation" />
+      )
+    case "informative":
+      return isProminent ? (
+        <Informative />
+      ) : (
+        <InformationWhiteIcon inheritSize role="presentation" />
+      )
+    case "negative":
+      return isProminent ? (
+        <Negative />
+      ) : (
+        <ExclamationWhiteIcon inheritSize role="presentation" />
+      )
+    case "positive":
+      return isProminent ? (
+        <Positive />
+      ) : (
+        <SuccessWhiteIcon inheritSize role="presentation" />
+      )
+    case "assertive":
+      return isProminent ? (
+        <Assertive />
+      ) : (
+        <ExclamationWhiteIcon inheritSize role="presentation" />
+      )
+  }
+}
+
+/**
+ * {@link https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082093114/Modal Guidance} |
+ * {@link https://cultureamp.design/?path=/docs/components-modals-confirmationmodal--docs Storybook}
+ */
+export const ConfirmationModal = ({
+  isOpen,
+  isProminent = false,
+  unpadded = false,
+  mood,
+  title,
+  onConfirm,
+  onAfterLeave,
+  confirmLabel = "Confirm",
+  dismissLabel = "Cancel",
+  confirmWorking,
+  automationId,
+  children,
+  ...props
```

Comment:
This appears to only contain `onDismiss`. Maybe just destructure it to reduce confusion (since the spread doesn't go anywhere - unless you want to add extra attributes, then it has a different purpose).